### PR TITLE
ID-3181 AWS SDK Version 3 + Modernised Build

### PR DESCRIPTION
**Pull request by promie**
This PR presents a comprehensive overhaul of changes aimed at modernising our build and bringing it up to date with current best practices. Specifically, we've achieved the following:

- Upgraded and refactored the codebase to utilise AWS SDK Version 3.
- Transitioned the project to TypeScript for improved type-safety and developer experience.
- Switched our build system to leverage `tsc` for compiling.
- Overhauled our testing suite, adopting Jest as our testing framework.
- Eliminated redundant dependencies, such as gulp and sinon, streamlining our dependency list to only essential packages.
- Completely rewritten both unit tests and acceptance tests, ensuring better coverage and reliability.
- Updated the README documentation, providing clear instructions on pre-requisites for new contributors.
- Mandated the use of Node.js version 18 and above, ensuring compatibility with the latest features and security patches.

Further bolstering our confidence in these changes, we have extensively tested this updated codebase. This includes not only publishing it onto npm but also integrating and utilising it within our primary project. All tests and integrations have yielded expected results, reassuring us of the robustness and reliability of this update.

It's worth noting that these changes represent a significant revamp, enhancing not just the functionality but also the maintainability and readability of our codebase.

**Note:** There's no rush in merging this pull request. Given the extensive changes, I understand the importance of reviewing it meticulously. Please take your time to go through it thoroughly and feel free to publish only when you're completely satisfied. I'm more than willing to make any necessary adjustments or modifications.

### CODE REVIEW

**Review by lucascanavan — APPROVED** [review]
Looks good to me @promie. @liamqma, we've done a much more through job this time. We believe this is now ready to merge and publish as v3 when you have time. Thanks!

**Comment by liamqma** [comment]
Nice work! @lucascanavan @promie  Thank you for modernising the repo. 

I'm a bit busy this week. I will review this PR this weekend.

File: package.json [hunk 1]
```diff
@@ -1,53 +1,51 @@
 {
-  "name": "beanstalkify",
+  "name": "elasticsprout",
```
Comment (liamqma) [inline package.json:2]: What's the reason to change the name here?

Comment (promie) [reply package.json]: Apologies @liamqma that was my bad. I was testing this project and forgot to change the name back. I've made a commit to rename it back to what it was.

**Review by liamqma — APPROVED** [review]

**Pull request merged**